boards/veer_el2_sim/README.md

```diff
@@ -0,0 +1,68 @@
+# VeeR EL2 simulation target
+
+For more information about the core and the Verilator testbench [visit the VeeR EL2 repository](https://github.com/chipsalliance/Cores-VeeR-EL2).
+This was tested with Verilator 5.006, which can be installed as explained [in the doc](https://verilator.org/guide/latest/install.html).
+
+There's also a repository dedicated to [running Tock on VeeR EL2 in simulation](https://github.com/chipsalliance/VeeR-EL2-tock-example)
+
+## Building software
+
+### Building Tock
+
+In order to compile Tock and convert it to the format used by Verilator, run:
+
+    make -C boards/veer_el2_sim debug -j$(nproc)
+    riscv64-unknown-elf-objcopy -O verilog target/riscv32imc-unknown-none-elf/debug/veer_el2_sim.elf kernel.hex
+
+### Building an application
+
+    git clone https://github.com/tock/libtock-c.git
+    make -C libtock-c/examples/c_hello TOCK_TARGETS='rv32imc|rv32imc.0x80030080.0x80070000|0x80030080|0x80070000' -j$(nproc)
+    riscv64-unknown-elf-objcopy -I binary -O verilog libtock-c/examples/c_hello/build/rv32imc/rv32imc.0x80030080.0x80070000.tbf c_hello.hex
+
+### Output files
+
+As the testbench for Verilator requires a single file with the program (`program.hex`), the kernel and the application need to be combined:
+
+    sed -i 's/@00000000/@80030000/g' c_hello.hex
```

Comment:
This address should be supported by libtock-c directly

Reply:
Here's an MR that adds support for the target in libtock-c: https://github.com/tock/libtock-c/pull/464
However, there is a tricky part when it comes to using the kernel with apps in the simulation testbench.

I believe the existing flow of swervolf doesn't address the issue of loading apps at all.

The testbench in VeeR uses a single file (`program.hex`) to initialize the memory.
The format of `program.hex` is the output of `objcopy -O verilog <...>`.

The kernel is built as an `.elf` file which provides the address and is understood by `objcopy` so it can be converted to the `.hex` file easily.
On the other hand, the application is built as `.tbf` which is not a format supported in `objcopy`. I don't know of any tool that would convert a `.tbf` to this format respecting the base address, so I've used `sed`.

So, even with the change in libtock-c to support the right addressing for VeeR this is needed.
I'll be happy to solve this differently but right now I think this is a limitation related to the TBF format.

Reply:
I've made changes in this PR, and also created the following PRs to implement the flow with tockloader:
https://github.com/tock/tockloader/pull/117
https://github.com/tock/libtock-c/pull/464
The readme explains the steps to simulate Tock with `c_hello`.